lib/Analysis/Allocation.cpp

```diff
@@ -267,10 +233,9 @@ class AllocationAnalysis {
       } else {
         auto smemShape = getRepShapeForAtomic(op->getResult(0));
         auto elems = getNumScratchElements(smemShape);
-        auto elemTy =
-            cast<triton::PointerType>(value.getType()).getPointeeType();
+        auto elemTy = cast<PointerType>(value.getType()).getPointeeType();
         auto bytes =
-            isa<triton::PointerType>(elemTy)
+            isa<PointerType>(elemTy)
```

Comment:
isn't the `isa` always true?

Reply:
We had the nested pointer type previously dropped by nvidia

Reply:
Let me try to remove it and see if there's still any use

Reply:
Ah, I think it's just a duplicate `isa` indeed, haha